### PR TITLE
Add support for portable configuration layouts

### LINKED ISSUE #1397: [BUG] CROSS_GetPlatformConfigDir function is not portable
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### What Operating System are you using?

Windows 10

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

CROSS_GetPlatformConfigDir always returns %LOCALAPPDATA%\DOSBox on Windows, even if user has config file in exe dir, thus shaders, soundfonts, etc are always loaded from %LOCALAPPDATA%, removing portability support. I believe this is an oversight.

### Code of Conduct & Contributing Guidelines

- [x] Yes, I agree.

### PR DESCRIPTION
**Pull request by kcgen**
If a `dosbox-staging.conf`[1] is found along-side the executable, then this directory is used as the base configuration path instead of the user's home directory path. This PR describes this in the README.

[1]  or `dosbox-staging-git.conf` for non-release binaries.

This allow Staging to be packaged and shared in a portable way, including custom GL shaders, MT-32 ROMs, and Soundfonts.

Thanks to @NicknineTheEagle for reporting this. Fixes #1397.

### CODE REVIEW

**Comment by NicknineTheEagle** [comment]
This does not account for the case where dosbox.conf is used.

**Comment by kcgen** [comment]
> This does not account for the case where dosbox.conf is used.

Using `dosbox.conf` strips the Staging identification from the configuration file and encourages confusion. I would expect that if a user has a Staging executable package, that it's not so much to ask that the conf be named `dosbox-staging.conf`.

ECE also uses a similar postfix: it's a good pattern that disambiguates settings for each fork.

When we read the conf from the user's home configuration directory, we also only read `dosbox-staging.conf` from there as well.  So in the same way, this portable behavior simply matches this.

Are there examples where this will cause users serious grief?  ie: a user who has a dosbox-staging installation for each game (Ok - then I can see the problem.. they will have to rename perhaps hundreds of conf files).

**Comment by kcgen** [comment]
@NicknineTheEagle , I'm going to merge this as-is.

If Staging parses `dosbox.conf` Staging-specific directories (ie: Staging's config path and Staging's binary path), the message we send users is that _"Staging can handle both kinds of conf files"_.  Which isn't the case; there are differences, and those are important to retain.  Likewise, we don't want users renaming Staging-specific conf files to `dosbox.conf`, and then expecting SVN to handle it.

Bottom line is I think it adds a lot of clarity for users that our conf files **_in Staging-specific paths_** be named Staging-specific.
